pkg/s3wrapper/handler.go

```diff
@@ -0,0 +1,56 @@
+package s3wrapper
+
+import (
+	"github.com/openshift/assisted-service/internal/isoeditor"
+	"github.com/openshift/assisted-service/internal/versions"
+	"github.com/sirupsen/logrus"
+)
+
+// NewStorageClient Create new storage client based off the inputs of DEPLOY_TARGET and STORAGE env variables
+func NewStorageClient(deployTarget string, storage string, s3cfg *Config, fsWorkDir string, log logrus.FieldLogger, versionsHandler versions.Handler, isoEditorFactory isoeditor.Factory) API {
+	var storageClient API
+	if storage != "" {
+		switch storage {
+		case "s3":
+			storageClient = NewS3Client(s3cfg, log, versionsHandler, isoEditorFactory)
+			if storageClient == nil {
+				log.Fatal("failed to create S3 client")
+			}
+		case "filesystem":
+			storageClient = NewFSClient(fsWorkDir, log, versionsHandler, isoEditorFactory)
+			if storageClient == nil {
+				log.Fatal("failed to create filesystem client")
+			}
+		default:
+			log.Fatalf("unsupported storage client: %s", storage)
+		}
+	} else {
+		switch deployTarget {
```

Comment:
I don't like that we distribute the logic of handling the deployTarget option in multiple places

Reply:
but maybe there is no other option?

Reply:
This is probably just for compatibility right? I would think that the different deployment scenarios should eventually know to set `STORAGE` correctly, but don't right now.

Reply:
This was just for the compatibility for the case of having STORAGE blank and a DEPLOY_TARGET. If STORAGE is set, then it overrides DEPLOY_TARGET.